src/foam/core/Model.js

```diff
@@ -85,6 +85,14 @@ foam.CLASS({
       // Is upgraded to an AxiomArray later.
       of: 'Method',
       name: 'methods'
+    },
+    {
+      class: 'Boolean',
+      name: 'queryObjectOnProjection',
+      documentation: `set to true if we cannot build object of the class's subclasses from projection results. 
+        e.g. the class's subclasses inherit properties, which require subclasses properties value for "calculation".
+        for example FixedFee 'formula' property requires 'fixedFee' property to calculate its result but 'fixedFee' property is not inherited from Fee class,
+        when using projection we cannot query fixedFee property of an object but we need it for 'formula' calculation when display Fee records from DAO`
     }
```

Comment:
what is the best place to place it?
should it be moved to some kind of menu configuration?

Reply:
Better to go in DAOMenu2.js I think.